github4s/src/main/scala/github4s/domain/Repository.scala

```diff
@@ -371,17 +371,34 @@ object RepoUrlKeys {
       files: Seq[FileComparison] = Seq.empty
   )
 
-  final case class FileComparison(
-      sha: String,
-      filename: String,
-      status: String,
-      additions: Int,
-      deletions: Int,
-      changes: Int,
-      blob_url: String,
-      raw_url: String,
-      contents_url: String,
-      patch: String
-  )
+  sealed trait FileComparison
```

Comment:
Would it make sense to add the fields in the trait too (as a `def`s)? I think it could be useful for handling the common fields

```scala
sealed trait FileComparison {
  def sha: String
  //...
}
```

Reply:
Done. Also added a couple helper methods for `getPatch` and `getPreviousFilename` for the cases where you don't necessarily want to do pattern matching.